### DISCUSSION TITLE
how to subset cells on a pycisTopic object?

### DISCUSSION THREAD

**Discussion by Melody-cell** [aertslab/pycisTopic#190] [General]
how can I subset cells on a pycisTopic object?
Did anyone know how to do it?

**Comment by SeppeDeWinter** [comment]
Hi @Melody-cell 

There is a `.subset` method that you can call on the object.

Best,

S

**Comment by Melody-cell** [comment]
hi, @SeppeDeWinter 
but how can i run qc on a after a "sample_id_to_barcodes_passing_filters" ?

**Comment by Melody-cell** [comment]
If i run qc and filters, then i run this:
![image](https://github.com/user-attachments/assets/c90a8e66-c1ad-4823-a786-a48f8ff68b48)
if i want to qc after filtered , should i change fragments.files?

**Reply by SeppeDeWinter** [reply]
Hi @Melody-cell 

Can you please clarify your question a bit further. I'm nut sure I'm following here ...

Best,

Seppe

**Reply by Melody-cell** [reply]
Hi @SeppeDeWinter , i mean if i want to use sample_id_to_barcodes_passing_filters to run QC step again,
( this step: 
!pycistopic qc \
    --fragments data/fragments.tsv.gz \
    --regions outs/consensus_peak_calling/consensus_regions.bed \
    --tss outs/qc/tss.bed \
    --output outs/qc/10x_multiome_brain )
how can i do it ?

**Reply by SeppeDeWinter** [reply]
Hi @Melody-cell 

I see, so you want to run the QC on a subset of cells?
In that case you have to use a filtered fragment file indeed.

But I'm not sure why you would want to do this? You can also subset the QC stats afterwards (for instance for visualisation purposes).

Best,

S

**Reply by Melody-cell** [reply]
Hi @SeppeDeWinter ,
OK, I see. Another question is that if i want to use the upper part of the tutorial as a qc tool, can I use just fragments files to do it without cell type annotations?
